browser/js/common/directives/cart/cart.js

```diff
@@ -1,8 +1,65 @@
 'use strict';
 
-app.directive('cart', function() {
-  return{
-		restrict: 'E',
-		templateUrl:'/js/common/directives/cart/cart.html'
-	}
-})
+app.directive('cart', function (CartFactory) {
+    return {
+        restrict: 'E',
+        templateUrl: '/js/common/directives/cart/cart.html',
+        link: function (scope) {
+            CartFactory.getItems()
+                .then(function (products) {
+                    // scope.products = products;
+                    var count = products.reduce(function (a, b) {
+                        return a + b.quantity
+                    }, 0)
+                    CartFactory.count = count
+                    scope.count = CartFactory.count
+                })
+        }
+    };
+});
+
+app.config(function ($stateProvider) {
+    $stateProvider.state('inCart', {
+        url: "/cart",
+        controller: "CartController",
+        templateUrl: "/js/common/directives/cart/incart.html",
+        resolve: {
+            products: function (CartFactory) {
+                return CartFactory.getItems()
```

Comment:
It would make sense to put products items in an object on the factory and reduce db calls